src/main/java/com/emc/pravega/perf/PravegaPerfTest.java

```diff
@@ -259,131 +314,278 @@ private static void parseCmdLine(String[] args) {
+                        key = Integer.toString(producerId + new Random().nextInt());
+                    } else {
+                        key = Integer.toString(producerId);
+                    }
+                   
                     // event ingestion
-                    long now = System.currentTimeMillis();
-                    retFuture = produceStats.runAndRecordTime(() -> {
-                                String key;
-                                if (isRandomKey) {
-                                    key = Integer.toString(producerId + new Random().nextInt());
-                                } else {
-                                    key = Integer.toString(producerId);
-                                }
+                    retFuture = stats.writeAndRecordTime(() -> {
                                 return fn.apply(key, payload);
                             },
-                            now,
-                            payload.length(), executor);
-                    //If it is a blocking call, wait for the ack
-                    if ( blocking ) {
-                        try {
-                            retFuture.get();
-                        } catch (InterruptedException  | ExecutionException e) {
-                            e.printStackTrace();
-                        }
-                    }
+                            payload.length());
 
                 }
+
                 long timeSpent = System.currentTimeMillis() - loopStartTime;
-                // wait for next event
-                try {
-                    //There is no need for sleep for blocking calls.
-                    if ( !blocking ) {
-                        if ( timeSpent < 1000) {
-                            Thread.sleep(1000 - timeSpent);
-                        }
-                    }
-                } catch (InterruptedException e) {
-                    // log exception
-                    System.exit(1);
-                }
-            }
+                     if (timeSpent < 1000) {
+                          Thread.sleep(1000 - timeSpent);
+                     }
+
+                diffTime = System.currentTimeMillis() - StartTime; 
+ 
+            } while(diffTime < mSeconds);
+
             producer.flush();
-            //producer.close();
-            try {
-                //Wait for the last packet to get acked
-                retFuture.get();
-            } catch (InterruptedException | ExecutionException e ) {
-                e.printStackTrace();
-            }
+            // producer.close();
+ 
+            //Wait for the last packet to get acked
+            retFuture.get();
         }
 
-        @Override
-        public void run() {
+      @Override
+      public Void call() throws Exception  {
             runLoop(sendFunction());
-            latch.countDown();
-        }
-    }
+            return null;   
+      }
+}
+
 
+class TransactionWriterWorker extends WriterWorker {
```

Comment:
static

Reply:
its not ;

Reply:
Please move this class into it's own file

Reply:
yes, its done